### PR TITLE
Fix cannot find control char when text length exceeds MAX_MENTION_QUERY_LENGTH

### PR DESCRIPTION
**Pull request by songkuixi**
* TL;DR: if `text.length > MAX_MENTION_QUERY_LENGTH`, the mentions will not working if the `MAX_MENTION_QUERY_LENGTH`'s char in original text is already attributed.
* In `HKWMentionsPluginV2`, when `controlCharactersToPrepend` is set, it'll trim the `parentTextView`'s text to be a substring of last `MAX_MENTION_QUERY_LENGTH` chars when trying to find the **last unattributed control char**. However, the `location` is **offset-ed** (`offset = originalText.length - MAX_MENTION_QUERY_LENGTH`) when passed to `mentionAttributeAtLocation` to determine whether the char at `location` in **original text** is already attributed. Before this fix, the `location`'s value range is `[0, trimmedText.length]` while it should be `[offset, trimmedText.length + offset]`. Thus we need to **add back the offset** if the string is trimmed.

* Example:

1. Original text: `<-- 200 chars, all attributed -->`
2. When adding a control char: `<-- 200 chars, all attributed --> @`
3. Trimmed text: `<-- 99 chars, all attributed --> @`
4. Use the trimmed text to find the last unattributed control char in original string:
```
<----- 200 chars, all attributed -----> @
<--- 99 chars ---> @                            <- before fix
                  ⬆️
<       offset      ><--- 99 chars ---> @       <- after fix
                                       ⬆️
```
Now it will think the last control char is attributed, but actually it's not.

Before|After
:-:|:-:
Simulator Screen Recording - iPhone 14 Pro - 2023-02-11 at 04 11 55|Simulator Screen Recording - iPhone 14 Pro - 2023-02-11 at 04 11 00

### CODE REVIEW

**Comment by songkuixi** [comment]
> should we lix this change?

Added a config property `enableControlCharacterMaxLengthFix`, default to `YES`. Will add a kill switch when integrating w/voyager.

**Review by li-jychen — APPROVED** [review]

**Review by dtykhyi — DISMISSED** [review]

**Review by li-jychen — APPROVED** [review]

**Review by mrsmiley009 — APPROVED** [review]
